apps/server/src/modules/h5p-editor/temporary-file-storage/temporary-file-storage.ts

```diff
@@ -67,35 +32,22 @@ export class TemporaryFileStorage implements ITemporaryFileStorage {
 
 	public async deleteFile(filename: string, userId: string): Promise<void> {
 		this.checkFilename(filename);
-		this.checkFilename(userId);
-		const filepath = join(this.path, userId, filename);
-		await rm(filepath);
-		await rm(`${filepath}.meta`);
-		const dir = dirname(filename);
-		if (dir !== '') {
-			const dirpath = join(this.path, userId, dir);
-			if ((await readdir(join(this.path, userId, dir))).length === 0) {
-				await rmdir(dirpath);
-			}
-		}
+		const meta = await this.repo.findByPath(userId, filename);
+		await this.s3Client.delete([join(userId, filename)]);
+		await this.repo.delete(meta);
 	}
 
 	public async fileExists(filename: string, user: IUser): Promise<boolean> {
 		this.checkFilename(filename);
-		this.checkFilename(user.id);
 		try {
-			await access(join(this.path, user.id, filename));
+			return !!(await this.repo.findByPath(user.id, filename));
```

Comment:
```suggestion
			await this.repo.findByPath(user.id, filename);
			return true;
```
This boolean conversion gives the impression that you rely on the return value of the repo method - but you don't. You just rely on it throwing an error.